m4/compiler.m4

```diff
@@ -2,6 +2,11 @@
 # -------------------------------------------------------------
 AC_DEFUN([LIBMESH_SET_COMPILERS],
 [
+  dnl We need to verify that we've done AC_ARG_ENABLE(mpi), AC_ARG_WITH(mpi), AC_ARG_ENABLE(petsc)
+  dnl which occur in LIBMESH_COMPILER_CONTROL_ARGS
+  AS_IF([test x"CALLED_COMPILER_CONTROL_ARGS" != x1],
+        [AC_MSG_ERROR([the compiler control args macro must be called before setting compilers])])
+
```

Comment:
Isn't this sort of thing what ```AC_REQUIRE([LIBMESH_COMPILER_CONTROL_ARGS])``` is designed to do, but more flexibly?

Reply:
It looks like you commented on an outdated commit here...indeed while it took me a while, I did come to `AC_REQUIRE` on my own 😆